### PR TITLE
feat: Homebrew

### PR DESCRIPTION
**Pull request by markphelps**
Fixes: FLI-417

- updates release workflow to do multi-platform release (+macOS)
- added [homebrew taps](https://goreleaser.com/customization/homebrew/) section in goreleaser release file to create the tap PR which should also allow us to run `flipt` as a service (still need to verify this works)

once released successfully, we can rename `.goreleaser.multi.yml` to just `.goreleaser.yml` and update all workflows

### CODE REVIEW

File: .goreleaser.multi.yml [hunk 1]
```diff
@@ -117,3 +117,27 @@ docker_manifests:
...
+
+    post_install: |
+      (var/"log/flipt").mkpath
+
+    service: |
+      run [opt_bin/"flipt"]
+      environment_variables PATH: std_service_path_env
+      keep_alive true
+      error_log_path "#{var}/log/flipt/server-stderr.log"
+      log_path "#{var}/log/flipt/server-stdout.log"
```
Comment (GeorgeMac) [inline .goreleaser.multi.yml:137]: I was looking at this going... what fresh hell of a configuration language is this.
Then I finally saw it. It's ruby, of course it is... Homebrew.

Comment (GeorgeMac) [reply .goreleaser.multi.yml:137]: yamlruby

Comment (GeorgeMac) [reply .goreleaser.multi.yml:137]: I was particularly perturbed by what the hell `opt_bin/` and `var/"flipt"` was.

Then I saw it and a quick GH search turned up the madness:
https://github.com/Homebrew/brew/blob/3c8b4949baefb1f8166749ff1a3f0665afadedda/Library/Homebrew/install_renamed.rb#L29-L31

Comment (markphelps) [reply .goreleaser.multi.yml:137]: yah a lot of it I copied from Pyroscope's brew formula: https://github.com/pyroscope-io/homebrew-brew/blob/main/Formula/pyroscope.rb#L44

**Review by GeorgeMac — APPROVED** [review]

**Review by yquansah — APPROVED** [review]

**Pull request merged**